bundles/org.openhab.binding.noolite/src/main/java/org/openhab/binding/noolite/handler/NooliteMTRF64BridgeHandler.java

```diff
@@ -0,0 +1,206 @@
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Dispose Noolite bridge handler {}", this.toString());
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Close bridge");
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Noolite bridge handler {}", this.toString());
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Initializing...");
+        bridgeConfig = getConfigAs(NooliteBridgeConfiguration.class);
+
+        // logger.debug("Checking Noolite adapter connection. {}", thing.getStatus());
+        if (thing.getStatus() != ThingStatus.ONLINE) {
+            connect();
+        }
+    }
+
+    @SuppressWarnings("null")
+    private void connect() {
+        logger.debug("Connecting to Noolite adapter");
+        try {
+            if (bridgeConfig.serial != null) {
+                if (adapter == null) {
+                    adapter = new NooliteMTRF64Adapter();
+                }
+            }
+            if (adapter != null) {
+                adapter.disconnect();
+                adapter.connect(bridgeConfig);
+            }
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Com-port error");
+            logger.debug(e.getMessage());
+        }
+    }
+
+    @SuppressWarnings({ "null", "unused" })
+    public void registerMegadThingListener(NooliteHandler thingHandler) {
+        String thingID = thingHandler.getThing().getConfiguration().get("type").toString() + "."
+                + (Integer.parseInt(thingHandler.getThing().getConfiguration().get("port").toString()));
+        if (thingHandlerMap.get(thingID) != null) {
+            thingHandlerMap.remove(thingID);
+        }
+        logger.debug("thingHandler for thing: '{}'", thingID);
+        if (thingHandlerMap.get(thingID) == null) {
+            thingHandlerMap.put(thingID, thingHandler);
+            logger.debug("register thingHandler for thing: {}", thingHandler);
+            updateThingHandlerStatus(thingHandler, this.getStatus());
+            if (thingID.equals("localhost.")) {
+                updateThingHandlerStatus(thingHandler, ThingStatus.OFFLINE);
+            }
+        } else {
+            logger.debug("thingHandler for thing: '{}' already registerd", thingID);
+            updateThingHandlerStatus(thingHandler, this.getStatus());
+        }
+    }
+
+    public ThingStatus getStatus() {
+        return getThing().getStatus();
+    }
+
+    private void updateThingHandlerStatus(NooliteHandler thingHandler, ThingStatus status) {
+        thingHandler.updateStatus(status);
+    }
+
+    @SuppressWarnings("null")
+    public static void updateValues(byte[] data) {
+        String thingID = data[1] + "." + data[4];
```

Comment:
This looks strange. Because the thingID will be 3 characters long. While the value put in the `thingHandlerMap` is type and port, which is much more information?